File: src/main/java/ninekothecat/catplugincore/money/interfaces/IBalanceHandler.java
package ninekothecat.catplugincore.money.interfaces;

import ninekothecat.catplugincore.money.enums.TransactionResult;

import java.util.UUID;

/**
 * the balance handler interface
 */
public interface IBalanceHandler {
    /**
     * does a transaction and returns the result
     *
     * @param transaction the transaction
     * @return the transaction result
     */
    @SuppressWarnings("UnusedReturnValue")
    TransactionResult doTransaction(ITransaction transaction);

     boolean userExists(UUID user);

     double getBalance(UUID user);

}

File: src/main/kotlin/ninekothecat/catplugincore/money/interfaces/IBalanceHandler.kt
package ninekothecat.catplugincore.money.interfaces

import ninekothecat.catplugincore.money.enums.TransactionResult
import java.util.*

/**
 * the balance handler interface
 */
interface IBalanceHandler {
    /**
     * does a transaction and returns the result
     *
     * @param transaction the transaction
     * @return the transaction result
     */
    fun doTransaction(transaction: ITransaction): TransactionResult
    fun userExists(user: UUID): Boolean
    fun getBalance(user: UUID): Double
}